src/dist_example.cpp

```diff
@@ -0,0 +1,223 @@
+  auto b = dist.exch(a, width);
+  OMEGA_H_CHECK(b == Reals({2.0 * reverse_root + 1, 2.0 * reverse_root}));
+}
+
+static void test_packet_fanout(CommPtr comm){
+  OMEGA_H_CHECK(comm->size() == 3);
+  Dist dist;
+  dist.set_parent_comm(comm);
+/*
+ * Starting Partition: {1, 2}  {0, 0}  {0, 0}
+ * End Partition:      {}      {1, 1}  {2, 2}
+*/
+
+  //Configure the directed graph. Note when explicitly setting parameters such as roots2items, 
+  //undefined behavior may occur when not set for every rank. Also note that roots2items is an
+  //offset array.
+  if(comm->rank() == 0){
+    dist.set_dest_ranks(Read<I32>({1, 1, 2, 2}));
+    dist.set_roots2items(LOs({0, 2, 4}));
+  }
+  else{
+    dist.set_dest_ranks(Read<I32>({})); 
+    //Offset arrays of length 0 may cause undefined behavior, especially in inverted dists
+    dist.set_roots2items(LOs({0}));
+  }
+
+  //Allocate Starting Partition
+  Reals a;
+  if(comm->rank() == 0)
+    a = Reals({1.0, 2.0});
+  else
+    a = Reals({});
+  
+  //Perform Exchange Operation
+  int width = 1;
+  auto b = dist.exch(a, width);
+  
+  //Check our that our end partition is what we expected
+  if(comm->rank() == 0)
+    OMEGA_H_CHECK(b == Reals({}));
+  else if(comm->rank() == 1){
+    OMEGA_H_CHECK(b == Reals({1, 1}));
+  }
+  else{
+    OMEGA_H_CHECK(b == Reals({2, 2}));
+  }
+}
+
+static void test_rank_fanout(CommPtr comm){
+  OMEGA_H_CHECK(comm->size() == 3);
+  Dist dist;
+  dist.set_parent_comm(comm);
+/*
+ * Starting Partition: {1, 2}  {0, 0}  {0, 0}
+ * End Partition:      {}      {1, 2}  {1, 2}
+*/
+
+  //Configure the directed graph. Note when explicitly setting parameters such as roots2items, 
+  //undefined behavior may occur when not set for every rank. Also note that roots2items is an
+  //offset array.
+  if(comm->rank() == 0){
+    dist.set_dest_ranks(Read<I32>({1, 2, 1, 2}));
+    dist.set_roots2items(LOs({0, 2, 4}));
+  }
+  else{
+    dist.set_dest_ranks(Read<I32>({}));
+    //Offset arrays of length 0 may cause undefined behavior, especially in inverted dists
+    dist.set_roots2items(LOs({0}));
+  }
+
+  //Allocate Starting Partition
+  Reals a;
+  if(comm->rank() == 0)
+    a = Reals({1.0, 2.0});
+  else
+    a = Reals({});
+
+  //Perform Exchange Operation
+  int width = 1;
+  auto b = dist.exch(a, width);
+
+  //Check End Partition is what we expected
+  if(comm->rank() == 0)
+    OMEGA_H_CHECK(b == Reals({}));
+  else{
+    OMEGA_H_CHECK(b == Reals({1, 2}));
+  }
+}
+
+static void test_exchange_reduce(CommPtr comm){
+  OMEGA_H_CHECK(comm->size() == 3);
+  Dist dist;
+  dist.set_parent_comm(comm);
+/*
+ * Starting Partition: {}   {1}  {1}
+ * End Partition:      {2}  {}   {}
+*/
+  
+  //Configure the directed graph. Note that dest_idxs must be set for all destination ranks, 
+  //as well as the number of destination roots on this rank.
+  if(comm->rank() == 0){
+    dist.set_dest_ranks(Read<I32>({}));
+    dist.set_dest_idxs(LOs({}), 1);
```

Comment:
we should describe these arguments to `set_dest_idxs`
- destination indices - empty array as rank 0 does not send
- reverse roots - there is one receiver (root) on this rank